src/arch/xtensa/up/xtos/reset-vector.S

```diff
@@ -179,11 +180,13 @@ _ResetHandler:
 
 #endif
 
-#if defined(CONFIG_APOLLOLAKE)
+#if defined(CONFIG_APOLLOLAKE) \
+	&& !(defined(CONFIG_SKYLAKE) || defined(CONFIG_KABYLAKE))
```

Comment:
This is really ugly and confusing...You are overloading the definition of CONFIG_APOLLOLAKE to a point where it's difficult to figure out what its definition is...